### ISSUE TITLE
mintapi 1.57 hangs on overview screen with --extended-transactions

### ISSUE THREAD

**Issue by apiology** [mintapi/mintapi#347]
**Describe the bug**
mintapi 1.57 hangs on overview screen

What version of MintAPI are you using? 1.57
What command[s] did you run / steps to reproduce?* mintapi --use-chromedriver-on-path --extended-transactions my-username my-password
Stacktrace/error received:

After waiting five minutes, I hit control-c  and get this:

```
C-c C-cTraceback (most recent call last):
  File "/Users/broz/.pyenv/versions/mylibs/bin/mintapi", line 8, in <module>
    sys.exit(main())
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 1380, in main
    data = mint.get_detailed_transactions(
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 835, in get_detailed_transactions
[O    df = pd.DataFrame(self.add_parent_category_to_result(result))
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 850, in add_parent_category_to_result
    parent = self.get_category_object_from_id(transaction['categoryId'])['parent']
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 1084, in get_category_object_from_id
    categories = self.get_categories()
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 985, in get_categories
    response = self.make_post_request(url=self.build_bundledServiceController_url(),
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 617, in make_post_request
    response = self.post(url=url, data=data, headers=JSON_HEADER)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/mintapi/api.py", line 614, in post
    return self.driver.request('POST', url, **kwargs)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/seleniumrequests/request.py", line 232, in request
    self.add_cookie(cookie_dict)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/seleniumrequests/request.py", line 154, in add_cookie
    super(RequestsSessionMixin, self).add_cookie(cookie_dict)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 894, in add_cookie
    self.execute(Command.ADD_COOKIE, {'cookie': cookie_dict})
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 319, in execute
    response = self.command_executor.execute(driver_command, params)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/selenium/webdriver/remote/remote_connection.py", line 374, in execute
    return self._request(command_info[0], url, body=data)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/selenium/webdriver/remote/remote_connection.py", line 397, in _request
    resp = self._conn.request(method, url, body=body, headers=headers)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/urllib3/request.py", line 78, in request
    return self.request_encode_body(
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/urllib3/request.py", line 170, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/urllib3/poolmanager.py", line 375, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/urllib3/connectionpool.py", line 719, in urlopen
    response = self.ResponseCls.from_httplib(
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/urllib3/response.py", line 597, in from_httplib
    headers = HTTPHeaderDict(headers.items())
  File "/Users/broz/.pyenv/versions/3.8.5/envs/mylibs/lib/python3.8/site-packages/urllib3/_collections.py", line 141, in __init__
    def __init__(self, headers=None, **kwargs):
KeyboardInterrupt

$
```

**What did you expect to happen?**

Fetch extended transaction data.

**What actually happened?**

Hangs for five minutes on accounts screen.  

**Additional context**

Reproducible headless in Docker (Linux) as well as on macOS.  Does not happen with 1.54.  Works fine on 1.57 without --extended-transactions flag.

![image](https://user-images.githubusercontent.com/3681194/142728533-625a8f54-d1b1-415e-b744-315f35d4cf3c.png)

**Comment by burkematthew** [comment]
@apiology can you include the exact command you used?

**Comment by burkematthew** [comment]
Thanks for the report @apiology .  It seems that the performance degradation is in an attempt to locate the parent category associated with the category of the individual transaction.  For the time being, try using either `--start-date` and/or `--end-date` to narrow the results that you are trying to pull while we identify a resolution for the performance degradation.

**Comment by apiology** [comment]
Sounds good!  Let me know if there's any more info I can provide in the meantime, and happy to retest when you're ready.

**Comment by burkematthew** [comment]
@apiology I will, thank you! I believe I've got the root cause identified, so I will keep you posted on the status of the fix.

**Comment by mrooney** [comment]
Resolved via #348, thanks all!
